src/core/Akka/Actor/Cancellation/Cancelable.cs

```diff
@@ -207,27 +206,17 @@ protected virtual void Dispose(bool disposing)
             // runtime from inside the finalizer and you should not reference
             // other objects. Only unmanaged resources can be disposed.
 
-            try
+            //Make sure Dispose does not get called more than once, by checking the disposed field
+            if(disposing)
             {
-                //Make sure Dispose does not get called more than once, by checking the disposed field
-                if(!_isDisposed)
+                if(!IsDisposed)
                 {
-                    if(disposing)
-                    {
-                        //Clean up managed resources
-                        if(_source != null)
-                        {
-                            _source.Dispose();
-                        }
-                    }
-                    //Clean up unmanaged resources
+                    //Clean up managed resources
+                    _source?.Dispose();
                 }
-                _isDisposed = true;
-            }
-            finally
-            {
-                // base.dispose(disposing);
+                //Clean up unmanaged resources
             }
+            IsDisposed = true;
```

Comment:
Why get rid of the `try...finally` here,  in case `source.Dispose()` throws?

Reply:
Hmmm... I can't see a CancellationTokenSource to throw inside its Dispose method, but i'll add the try...catch block back

Reply:
To my defense, the original code does not catch any exception thrown from Dispose, it only have an empty finally block